sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -292,14 +294,15 @@ export class Agent<ContentTypes = unknown> extends EventEmitter<
     }
   }
 
-  async start() {
+  async start(options?: StreamOptions<Message, DecodedMessage<ContentTypes>>) {
```

Comment:
i think we can remove the generic values and the `onValue` field for this since we're providing it.

so something like this:
```
export type AgentStreamOptions = Omit<StreamOptions, "onValue">;
```

WDYT?